runtime/compiler/optimizer/abstractinterpreter/AbsInterpreter.cpp

```diff
@@ -0,0 +1,2848 @@
+   }
+
+void TR::AbsBlockInterpreter::constant(int64_t l)
+   {
+   TR::AbsStackMachineState* state = getState();
+   TR::AbsValue* value1 = createLongConst(l);
+   TR::AbsValue* value2 = createTopLong();
+   state->push(value1);
+   state->push(value2);
+   }
+
+void TR::AbsBlockInterpreter::constant(float f)
+   {
+   TR::AbsStackMachineState* state = getState();
+   TR::AbsValue* floatConst = createTopFloat();
+   state->push(floatConst);
+   }
+
+void TR::AbsBlockInterpreter::constant(double d)
+   {
+   TR::AbsStackMachineState* state = getState();
+   TR::AbsValue *value1 = createTopDouble();
+   TR::AbsValue *value2 = createTopDouble();
+   state->push(value1);
+   state->push(value2);
+   }
+
+void TR::AbsBlockInterpreter::constantNull()
+   {
+   TR::AbsStackMachineState* state = getState();
+   TR::AbsValue* value = createNullObject();
+   state->push(value);
+   }
+
+void TR::AbsBlockInterpreter::ldc(bool wide)
+   {
+   TR::AbsStackMachineState* state = getState();
+
+   int32_t cpIndex = wide ? _bci.next2Bytes() : _bci.nextByte();
+   TR::DataType type = _callerMethod->getLDCType(cpIndex);
+
+   switch (type)
+      {
+      case TR::Int32:
+         {
+         int32_t intVal = _callerMethod->intConstant(cpIndex);
+         constant((int32_t)intVal);
+         break;
+         }
+      case TR::Int64:
+         {
+         int64_t longVal = _callerMethod->longConstant(cpIndex);
+         constant((int64_t)longVal);
+         break;
+         }
+      case TR::Double:
+         {
+         constant((double)0); //the value does not matter here since doubles are modeled as TOP
+         break;
+         }
+      case TR::Float:
+         {
+         constant((float)0); //same for float
+         break;
+         }
+      case TR::Address:
+         {
+         if (_callerMethod->isStringConstant(cpIndex))
+            {
+            TR::SymbolReference *symbolReference = comp()->getSymRefTab()->findOrCreateStringSymbol(_callerMethodSymbol, cpIndex);
+            if (symbolReference->isUnresolved())
+               {
+               state->push(createTopObject());
+               }
+            else  //Resolved
+               {
+               TR::AbsValue *stringVal = createStringObject(symbolReference, true);
+               state->push(stringVal);
+               }
+            break;
+            }
+         else if (_callerMethod->isClassConstant(cpIndex))
+            {
+            TR_OpaqueClassBlock* classBlock = _callerMethod->getClassFromConstantPool(comp(), cpIndex);
+            if (classBlock)
+               {
+               TR_OpaqueClassBlock* jlClass = comp()->fe()->getClassClassPointer(classBlock);
+               TR::AbsValue* value = createNonNullObject(classBlock);
+               state->push(value);
+               }
+            else
+               {
+               state->push(createNullObject());
+               }
+            break;
+            }
+
+         state->push(createTopObject());
```

Comment:
If this is a condy of type `long` or `double`, then it should push two entries to the stack. At this point I don't want to ask you to detect that case or even detect condy. Since the possibilities other than string and class are all somewhat `MethodHandle`-related, I think it would be fine for now for this path to cause `interpretByteCode()` to fail